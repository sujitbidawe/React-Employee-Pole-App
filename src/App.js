import './App.css';
import { useEffect, Fragment } from "react";
import { connect } from "react-redux";
import { Routes, Route } from 'react-router-dom';
import { handleInitialData } from './actions/shared';
import Dashboard from './components/Dashboard';
import Login from './components/Login';
import ProtectedRoute from './components/ProtectedRoute';
import LoadingBar from "react-redux-loading-bar";

const App = (props) => {
	useEffect(() => {
		  props.dispatch(handleInitialData());
	}, [])

	return <div>
		<Fragment>
			<LoadingBar />
			<div className="container">
				{
					props.loading ? null :
					<Routes>
						<Route index element={<Login />} />
						<Route exact path="login" element={<Login />} />
						<Route element={<ProtectedRoute user={props.authedUser} />}>
							<Route exact path="dashboard" element={<Dashboard />} />
						</Route>
						<Route path="*" className='container' element={<p>There's nothing here: 404!</p>} />
					</Routes>
				}
			</div>
		</Fragment>
	</div>;
};

const mapStateToProps = ({authedUser, users}) => ({
	loading: JSON.stringify(users) === JSON.stringify({}),
	authedUser
})

export default connect(mapStateToProps)(App);