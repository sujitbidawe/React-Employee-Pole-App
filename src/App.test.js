import { render } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from "react-router-dom";
import reducer from './reducers'
import middleware from "./middleware";
import { createStore } from 'redux';

const store = createStore(reducer, middleware);

test('renders learn react link', () => {
  var view =render(
	<Provider store={store}>
        <Router>
            <App />
        </Router>
    </Provider>
  );
  expect(view).toMatchSnapshot();
});
