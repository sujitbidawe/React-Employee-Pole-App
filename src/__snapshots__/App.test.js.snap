// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="direction: ltr;"
        >
          <div
            class=""
            style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
          />
          <div
            style="display: table; clear: both;"
          />
        </div>
        <div
          class="container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <div
        class="container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
